services/dynamic/dynamic-xml.service.js

```diff
@@ -47,14 +36,50 @@ module.exports = class DynamicXml extends BaseService {
     } catch (e) {
       throw new InvalidParameter({ prettyMessage: e.message })
     }
-    values = values.map((node, i) =>
-      pathIsAttr ? node.value : node.firstChild.data
-    )
+
+    if (!Array.isArray(values)) {
+      throw new InvalidResponse({
+        prettyMessage: 'must be a collection of elements',
+      })
+    }
+
+    values = values.reduce((accum, node) => {
+      if (pathIsAttr) {
+        accum.push(node.value)
+      }
+
+      if (node.firstChild) {
+        accum.push(node.firstChild.data)
+      }
+
+      return accum
```

Comment:
One of the runtime errors was trying to access `data` field when `firstChild` was null/undefined. My thinking is that it would be better to just ignore the node if all the conditions are false vs. throwing an exception. I'm not sure if that is the "correct" behavior, but it should get rid of the runtime error either way.